src/pip/_internal/resolution/resolvelib/candidates.py

```diff
@@ -291,6 +299,26 @@ def __init__(
             version=version,
         )
 
+    def iter_dependencies(self):
+        # type: () -> Iterable[Optional[Requirement]]
+        dist = None  # type: Optional[Distribution]
+        preparer, lazy_wheel = self._factory.preparer, self._factory.lazy_wheel
+        remote_wheel = self._link.is_wheel and not self._link.is_file
+        if lazy_wheel and remote_wheel and not preparer.require_hashes:
+            assert self._name is not None
+            logger.info('Collecting %s', self._ireq.req or self._ireq)
+            # If RuntimeError is raised, fallback to self.dist.
+            with indent_log(), suppress(RuntimeError):
```

Comment:
Why would a Runtime Error be raised here?

Reply:
This is raised from the initialization of the lazy wheel object if range requests are not supported by the package index:

https://github.com/pypa/pip/blob/c2350444e1b32801a191cac3b8d9ed9026c48d94/src/pip/_internal/network/lazy_wheel.py#L62-L63

I don't think RuntimeError is descriptive though so I'm open for suggestion.

Reply:
I'd suggest raising a specific subclass and instead of suppressing it, to logger.debug when that happens.

Reply:
ad5d3dc94e9dcbbb749bb350ba0c958cbd57279f used `HTTPRangeRequestUnsupported`.  Do you think that name is a bit too long?  Also, how about the log message?

https://github.com/pypa/pip/blob/ad5d3dc94e9dcbbb749bb350ba0c958cbd57279f/src/pip/_internal/resolution/resolvelib/candidates.py#L322-L326

Reply:
HTTPRangeRequestUnsupported is a good exception name